community/cypher/cypher-compiler-3.0/src/main/scala/org/neo4j/cypher/internal/compiler/v3_0/mutation/DeleteEntityAction.scala

```diff
@@ -72,10 +72,15 @@ case class DeleteEntityAction(elementToDelete: Expression, forced: Boolean)
   def symbolTableDependencies = elementToDelete.symbolTableDependencies
 
   def localEffects(symbols: SymbolTable) = elementToDelete match {
-    case i: Variable => symbols.variables(i.entityName) match {
-      case _: NodeType         => Effects(DeletesNode)
-      case _: RelationshipType => Effects(DeletesRelationship)
-      case _: PathType         => Effects(DeletesNode, DeletesRelationship)
+    case i: Variable => effectsFromCypherType(symbols.variables(i.entityName))
+    case ContainerIndex(i: Variable, _) => symbols.variables(i.entityName) match {
```

Comment:
Just tested it; nested collection is supported, so the `innerType` here can also be `ContainerIndex`. This needs to be recursive.